server/mediapublic/mediapublic/views.py

```diff
@@ -594,8 +596,138 @@ def blogs_comment_by_id(request):
+        request.route_url('edit_user_playlist', uid=person_id, pid=playlist['id'])
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        request.response.status = 405
+    return resp
+
+# [GET]
+@view_config(route_name='edit_user_playlist', renderer='json')
+def edit_user_playlist(request):
+    cls = Playlists
+    if request.method == "GET":
+        playlist_id = request.matchdict['id']
+        playlist = Playlists.get_by_id(playlist_id)
+        # Not using do_get_single because using sqlalchemy relationships from actual object, not dict
+        chosen_recordings = playlist.recordings
+        chosen_ids = [rec.id for rec in chosen_recordings]
+
+        unchosen_recordings = DBSession.query(
+            Recordings,
+        ).filter(
+            Recordings.id not in chosen_ids
+        )
+        # recordings user has in playlist vs recordings that can be added to it
+        # might want to limit unchosen, maybe take random 10
+
+        resp = {
+            'chosen_recordings': chosen_recordings,
+            'unchosen_recordings': unchosen_recordings
+        }
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        request.response.status = 405
+    return resp
+
+
+# [GET]
+@view_config(route_name='new_user_playlist', renderer='json')
+def new_user_playlist(request):
+    # This should just return a form for title of new playlist.
+    resp = {}
+    if request.method == "GET":
+        resp = {'author_id': request.matchdict['uid']}
+        # pass through form
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        request.response.status = 405
+    return resp
+
+# [           PUT        ]
+@view_config(route_name='assign_to_playlist')
+def assign_to_playlist(request):
+    cls = PlaylistAssignments
+    resp = {}
+    status = 500
+    if request.method == "PUT":
+        keys = {
+            'playlist_id': request.matchdict['id'],
+            'recording_id': request.body['recording_id'],
+            # allowed to use request.body? check_payload wouldn't work for this
+        }
+        resp, status = do_post(request, keys, cls)
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        status = 405
+    request.response.status = status
+    return resp
+
+# [           PUT        ]
+@view_config(route_name='remove_from_playlist')
+def remove_from_playlist(request):
+    cls = PlaylistAssignments
+    resp = {}
+    status = 500
+    if request.method == "PUT":
+        pa_id =
+            PlaylistAssignments.get_by_playlist_id_and_recording_id(
+                cls, request.matchdict['id'], request.body['recording_id']
+            ).id
+        # reverse search isn't a part of do_delete
+
+        resp, status = do_delete(request, pa_id, cls)
+    else:
+        resp = {'error': 'Method Not Allowed'}
+        status = 405
+    request.response.status = status
+    return resp
+
+
+
+# [GET,       PUT, DELETE]
+@view_config(route_name='user_playlist', renderer='json')
```

Comment:
I would have expected this function to look almost identical to this one:

https://github.com/ismcodes/mediapublic/blob/playlists/server/mediapublic/mediapublic/views.py#L250

It's okay to change the defined scheme ... but it needs to change everywhere then.